src/Data/Document/DocumentEditingConfig.ts

```diff
@@ -1,17 +1,19 @@
 import { provideEditingConfig } from 'scrivito'
-import { Document } from './DocumentDataClass'
+import { DocumentPromise } from './DocumentDataClass'
 
-provideEditingConfig(Document, {
-  title: 'Document',
-  attributes: {
-    _id: { title: 'Document ID' },
-    title: { title: 'Title' },
-    number: { title: 'Number' },
-    type: { title: 'Type' },
-    createdAt: { title: 'Created at' },
-    language: { title: 'Language' },
-    format: { title: 'Format' },
-    size: { title: 'Size' },
-    version: { title: 'Version' },
-  },
+DocumentPromise.then((Document) => {
```

Comment:
* [ ] FYI Our compiler setup allows for top-level await, so we could write this in a more readable form:

```tsx
provideEditingConfig(await DocumentPromise, {
  // ...
})

// or

const Document = await DocumentPromise
provideEditingConfig(Document, {
  // ...
})
```

Reply:
Sadly not in all environments:

```
x Build failed in 3.32s
error during build:
[vite:esbuild-transpile] Transform failed with 14 errors:
assets/editingConfigs-!~{005}~.js:30:17: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
assets/editingConfigs-!~{005}~.js:46:42: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
assets/editingConfigs-!~{005}~.js:67:42: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
assets/editingConfigs-!~{005}~.js:74:42: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
assets/editingConfigs-!~{005}~.js:91:42: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
...

Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
28 |  });
29 |
30 |  const Document = await DocumentPromise;
   |                   ^
31 |  index_browserExports.provideEditingConfig(Document, {
32 |    title: "Document",
```